src/transformStateWithClones.js

```diff
@@ -2,12 +2,40 @@
 
 /**
  * @param {Object} state
- * @param {Object[]} actions
+ * @param {Object[]} actionions
  *
  * @return {Object[]}
  */
-function transformStateWithClones(state, actions) {
-  // write code here
+function transformStateWithClones(state, actionions) {
+  const finalArray = [];
+  const newObj = { ...state };
+
+  for (const action of actionions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(newObj, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete newObj[key];
+        }
+        break;
+
+      case 'clear':
+        for (const stateKey in newObj) {
```

Comment:
this is why we should give proper names, this loop looks confusing
why do we search for a state key in the new object? what is the new object?